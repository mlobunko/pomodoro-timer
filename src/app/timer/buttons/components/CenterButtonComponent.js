import React from "react";
import PropTypes from "prop-types";

import Button from "shared/button/components";

export const CenterButtonComponent = ({ pauseTimer, startTimer, timerOn }) => (
  <React.Fragment>
    {timerOn ? (
      <Button icon="pause" onClick={pauseTimer} />
    ) : (
      <Button icon="play" onClick={startTimer} />
    )}
  </React.Fragment>
);

CenterButtonComponent.displayName = "CenterButton";

export default CenterButtonComponent;

CenterButtonComponent.propTypes = {
  pauseTimer: PropTypes.func.isRequired,
  startTimer: PropTypes.func.isRequired,
  timerOn: PropTypes.bool.isRequired
};
