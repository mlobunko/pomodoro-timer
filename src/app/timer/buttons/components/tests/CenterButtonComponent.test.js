import React from "react";
import { shallow } from "enzyme";

import { CenterButtonComponent } from "../CenterButtonComponent";

const props = {
  pauseTimer: jest.fn(),
  startTimer: jest.fn(),
  timerOn: true
};

describe("props", () => {
  describe("timerOn true", () => {
    it("button pause onClick is 'pauseTimer'", () => {
      const wrapper = shallow(<CenterButtonComponent {...props} />);
      expect(wrapper.find("[icon='pause']").prop("onClick")).toEqual(
        props.pauseTimer
      );
    });
  });

  describe("timerOn false", () => {
    it("button play onClick is 'startTimer'", () => {
      const wrapper = shallow(
        <CenterButtonComponent {...props} timerOn={false} />
      );
      expect(wrapper.find("[icon='play']").prop("onClick")).toEqual(
        props.startTimer
      );
    });
  });
});

describe("render", () => {
  it("renders correctly, timerOn true", () => {
    const wrapper = shallow(<CenterButtonComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, timerOn false", () => {
    const wrapper = shallow(
      <CenterButtonComponent {...props} timerOn={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
