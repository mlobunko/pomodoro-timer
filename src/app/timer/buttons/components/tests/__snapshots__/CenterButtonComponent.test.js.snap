// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly, timerOn false 1`] = `
<Fragment>
  <Button
    icon="play"
    onClick={[Function]}
  />
</Fragment>
`;

exports[`render renders correctly, timerOn true 1`] = `
<Fragment>
  <Button
    icon="pause"
    onClick={[Function]}
  />
</Fragment>
`;
